state/types.go

```diff
@@ -70,18 +69,10 @@ type Service struct {
 	Meta
 }
 
-// Identifier returns the endpoint key name or ID.
-func (s1 *Service) Identifier() string {
-	if s1.Name != nil {
-		return *s1.Name
-	}
-	return *s1.ID
-}
-
```

Comment:
Do we need to actually remove these? They were exported, so this is breaking change to the deck.state package, even if other deck packages no longer use it and deck itself no longer needs them. 

I don't know if anything else was actually using them (the controller isn't based on brief review) and they are probably just internal deck utils in practice that were only exported because we need them in another deck package, but IMO the proper cautious approach here would be to deprecate with instructions to switch to FriendlyName, then remove entirely in the next major.

The less-cautious approach is probably fine here though given that `FriendlyName()` is a drop-in replacement and that the chance of actual usage outside is very low. That said, we should (ignoring that I have merged things today without it) probably switch to the changelog model we've since adopted in other repos: we add an unreleased section for the next release in the changelog and update it in individual PRs. I guarantee that we'd probably miss this (admittedly minor and irrelevant to end users) breaking change if we write it prior to release, since it's buried in the code and not readily apparent from the overall purpose of the change.

I'd recommend we go with the update changelog now given that it's simpler than reverting the removals and the expected impact of the breaking change is minimal.

Reply:
You're right, we'd probably just want to leave these untouched to lower the (already low) likelihood of braking something using these.